File: process-management-api/src/main/java/com/victorsantos/processmanagementapi/users/usercases/admin/updateuser/UpdateUserUserCaseRequest.java
package com.victorsantos.processmanagementapi.users.usercases.admin.updateuser;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Data
public class UpdateUserUserCaseRequest {
  private String id;
  private String name;
  private String email;
  private String role;
}

File: process-management-web/src/users/admin/usercases/update-user/UpdateUserUserCaseRequest.ts
import { UserRole } from "../../../../security/UserRole";

export default class UpdateUserUserCaseRequest {
  id: string;
  name: string;
  email: string;
  role: UserRole;

  constructor(attrs: {
    id: string;
    name: string;
    email: string;
    role: UserRole;
  }) {
    this.id = attrs.id;
    this.name = attrs.name;
    this.email = attrs.email;
    this.role = attrs.role;
  }
}
